mqtt/mqtt3/src/client/mod.rs

```diff
@@ -520,6 +520,7 @@ pub enum Event {
 pub enum SubscriptionUpdateEvent {
     Subscribe(crate::proto::SubscribeTo),
     Unsubscribe(String),
+    SubscriptionRejectedByServer(String),
```

Comment:
What was mqtt3 doing before if it was rejected?

Reply:
reconnecting the client and retry to send the subscription, no event was sent to caller